syntax = "proto2";
package oneflow;

message CudaPinnedMemory {
  required int64 device_id = 1;
}

message FakeDevicePinnedMemory {
  optional int64 reserved = 1;
}

message HostMemory {
  oneof page_lock_case {
    CudaPinnedMemory cuda_pinned_mem = 1;
    FakeDevicePinnedMemory fake_dev_pinned_mem = 3;
  }
  optional bool used_by_network = 2 [default = false];
}

message DeviceCudaMemory {
  required int64 device_id = 1;
}

message FakeDeviceMemory {
  optional int64 reserved = 1;
}

message DeviceAbcMemory {
  optional int64 reserved = 1;
}

message DeviceEnflameMemory {
  required int64 device_id = 1;
}

message MemoryCase {
  oneof case {
    HostMemory host_mem = 1;
    DeviceCudaMemory device_cuda_mem = 2;
    FakeDeviceMemory fake_dev_mem = 3;
    DeviceEnflameMemory device_enflame_mem = 4;
    DeviceAbcMemory device_abc_mem = 5;
  }
}
